dodo.c

```diff
@@ -372,10 +375,31 @@ struct Instruction * parse_line(char *source, size_t *index){
         return 0;
     }
 
-    free(i);
+    /* ln where n is positive integer */
+    switch( source[*index] ){
+        case 'l':
+        case 'L':
+            ++(*index);
+            break;
+        default:
+            printf("parse_line: unexpected character '%c', expected 'l'\n", source[*index]);
+            free(i);
+            return 0;
+            break;
+    }
+
+    ret = parse_number(i, source, index);
+
+    if( i->argument.num == 0 ){
```

Comment:
note that in the case of the call to `parse_number` failing the state of `i->argument.num` may not be known, we *know* that `new_instruction(LINE)` uses calloc, but this may not always be true... we may have to think about this carefully.

Currently the worst case I think of is that we will see this puts error printed in cases where it shouldn't be, which isn't a bad case.